src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/SymbolEquivalenceComparer.EquivalenceVisitor.cs

```diff
@@ -45,17 +45,18 @@ public bool AreEquivalent(ISymbol x, ISymbol y, Dictionary<INamedTypeSymbol, INa
                     throw new InvalidOperationException("Stack too deep.");
                 }
 #endif
+
                 // https://github.com/dotnet/roslyn/issues/39643 This is a temporary workaround sufficient to get existing tests passing.
                 // This component should be modified to properly deal with differences caused by nullability.
                 if (x is ITypeSymbol xType && y is ITypeSymbol yType && xType.IsDefinition != yType.IsDefinition)
                 {
                     if (x.IsDefinition)
                     {
-                        y = yType.WithNullableAnnotation(xType.NullableAnnotation);
+                        y = yType.WithNullableAnnotationFrom(xType);
```

Comment:
New extension method `WithNullableAnnotationFrom` performs the exact same functionality when linked into Workspaces layer, but is a no-op in CodeStyle layer, which still does not have access to very recently added public APIs `WithNullableAnnotation` and `NullableAnnotation`.